### PR TITLE
feat: Add `CachedSetValuer`

### PR DESCRIPTION
**Pull request by pblivin0x**

### CODE REVIEW

File: contracts/product/CachedSetValuer.sol [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+    /**
+     * Returns both cached and current valuations for comparison.
+     * Does not modify state or update cache.
+     * 
+     * @param _setToken        SetToken to get valuation for
+     * @param _quoteAsset      Asset to quote valuation in
+     * @return ValuationInfo   Struct containing cached and current valuations
+     */
+    function previewValuation(
```
Comment (edkim) [inline contracts/product/CachedSetValuer.sol:186]: Is the `currentValue` returned by the setValuer not 100% accurate without first calling sync on the rebasing module?

**Review by edkim — COMMENTED** [review]
Code looks good to me. Did have one question on the `previewValuation` function to confirm my understanding.

File: contracts/product/CachedSetValuer.sol (hunk 1 above)
Comment (pblivin0x) [reply contracts/product/CachedSetValuer.sol:187]: yes this is correct - added a warning here [`6b24709`](https://github.com/IndexCoop/index-protocol/pull/51/commits/6b24709a0e15ad587d91d05cf835de5fb420d28f)

this function could be updated to simulate the accounting update while still not writing to state

**Review by edkim — APPROVED** [review]
🚢

**Pull request merged**